front/src/modules/simulationResult/components/ChartHelpers/drawCurve.ts

```diff
@@ -59,4 +60,60 @@ const drawCurve = <
   }
 };
 
+export const drawCurveV2 = <
```

Comment:
I think you can remove drawCurveV2 and only modify the type of T in drawCurve like this
```
const drawCurve = <
  T extends
    | ArrayElement<GevPreparedData[keyof GevPreparedData]>
    | ArrayElement<
        GevPreparedDataV2[keyof Omit<GevPreparedDataV2, 'electrifications' | 'pathLength'>]
      >
    | ConsolidatedPosition,
>(
```

Reply:
Same here, I thought it would be easier, when dropping v1, to just remove the all function than go look in types how it works. I will add a '// TODO DROPV1' for the dracCurve function so we don't forget to remove it if it's ok for you.